ccl/logictestccl: TestCCLLogic failed

[(ccl/logictestccl).TestCCLLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2622732&tab=buildLog) on [master@1a459a81dba35b6a091f0a2954aa33d50f1e5d24](https://github.com/cockroachdb/cockroach/commits/1a459a81dba35b6a091f0a2954aa33d50f1e5d24):

```
               ),
               INDEX i2 (w ASC) PARTITION BY LIST (w) (
                 PARTITION p3 VALUES IN ((5), (6)),
                 PARTITION p4 VALUES IN ((7), (8))
               ),
               FAMILY fam_0_x_y_z_w (x, y, z, w)
            );
            ALTER PARTITION p1 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15210;
            ALTER PARTITION p2 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15213;
            ALTER PARTITION p3 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15411;
            ALTER PARTITION p4 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15418
        but found (query options: "") :
            t  CREATE TABLE public.t (
               x INT8 NOT NULL,
               y INT8 NOT NULL,
               z INT8 NULL,
               w INT8 NULL,
               CONSTRAINT "primary" PRIMARY KEY (y ASC),
               UNIQUE INDEX t_x_key (x ASC),
               INDEX i1 (z ASC) PARTITION BY LIST (z) (
                 PARTITION p1 VALUES IN ((1), (2)),
                 PARTITION p2 VALUES IN ((3), (4))
               ),
               INDEX i2 (w ASC) PARTITION BY LIST (w) (
                 PARTITION p3 VALUES IN ((5), (6)),
                 PARTITION p4 VALUES IN ((7), (8))
               ),
               FAMILY fam_0_x_y_z_w (x, y, z, w)
            );
            ALTER PARTITION p3 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15411;
            ALTER PARTITION p4 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15418;
            ALTER PARTITION p1 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15210;
            ALTER PARTITION p2 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15213
--- done: testdata/logic_test/zone with config 5node: 144 tests, 1 failures
--- total progress: 571 statements/queries
            --- FAIL: TestCCLLogic/5node/zone/authorization (1.27s)
=== RUN   TestCCLLogic/5node/zone
=== PAUSE TestCCLLogic/5node/zone
=== CONT  TestCCLLogic/5node/zone
        --- FAIL: TestCCLLogic/5node/zone (3.05s)
=== RUN   TestCCLLogic/5node
    --- FAIL: TestCCLLogic/5node (0.00s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestCCLLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/logictestccl).TestCCLLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2625988&tab=buildLog) on [master@1413f686db7b70a3b25cf336baa283a7767027e1](https://github.com/cockroachdb/cockroach/commits/1413f686db7b70a3b25cf336baa283a7767027e1):

```
               ),
               INDEX i2 (w ASC) PARTITION BY LIST (w) (
                 PARTITION p3 VALUES IN ((5), (6)),
                 PARTITION p4 VALUES IN ((7), (8))
               ),
               FAMILY fam_0_x_y_z_w (x, y, z, w)
            );
            ALTER PARTITION p1 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15210;
            ALTER PARTITION p2 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15213;
            ALTER PARTITION p3 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15411;
            ALTER PARTITION p4 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15418
        but found (query options: "") :
            t  CREATE TABLE public.t (
               x INT8 NOT NULL,
               y INT8 NOT NULL,
               z INT8 NULL,
               w INT8 NULL,
               CONSTRAINT "primary" PRIMARY KEY (y ASC),
               UNIQUE INDEX t_x_key (x ASC),
               INDEX i1 (z ASC) PARTITION BY LIST (z) (
                 PARTITION p1 VALUES IN ((1), (2)),
                 PARTITION p2 VALUES IN ((3), (4))
               ),
               INDEX i2 (w ASC) PARTITION BY LIST (w) (
                 PARTITION p3 VALUES IN ((5), (6)),
                 PARTITION p4 VALUES IN ((7), (8))
               ),
               FAMILY fam_0_x_y_z_w (x, y, z, w)
            );
            ALTER PARTITION p3 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15411;
            ALTER PARTITION p4 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15418;
            ALTER PARTITION p1 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15210;
            ALTER PARTITION p2 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15213
--- done: testdata/logic_test/zone with config 5node: 144 tests, 1 failures
--- total progress: 579 statements/queries
            --- FAIL: TestCCLLogic/5node/zone/authorization (2.46s)
=== RUN   TestCCLLogic/5node/zone
=== PAUSE TestCCLLogic/5node/zone
=== CONT  TestCCLLogic/5node/zone
        --- FAIL: TestCCLLogic/5node/zone (5.93s)
=== RUN   TestCCLLogic/5node
    --- FAIL: TestCCLLogic/5node (0.00s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestCCLLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/logictestccl).TestCCLLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2626332&tab=buildLog) on [master@86c619f768a4d6f4cc80efb5c9479e5fc0fd72c1](https://github.com/cockroachdb/cockroach/commits/86c619f768a4d6f4cc80efb5c9479e5fc0fd72c1):

```
                 PARTITION p2 VALUES IN ((3), (4))
               ),
               INDEX i2 (w ASC) PARTITION BY LIST (w) (
                 PARTITION p3 VALUES IN ((5), (6)),
                 PARTITION p4 VALUES IN ((7), (8))
               ),
               FAMILY fam_0_x_y_z_w (x, y, z, w)
            );
            ALTER PARTITION p1 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15210;
            ALTER PARTITION p2 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15213;
            ALTER PARTITION p3 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15411;
            ALTER PARTITION p4 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15418
        but found (query options: "") :
            t  CREATE TABLE public.t (
               x INT8 NOT NULL,
               y INT8 NOT NULL,
               z INT8 NULL,
               w INT8 NULL,
               CONSTRAINT "primary" PRIMARY KEY (y ASC),
               UNIQUE INDEX t_x_key (x ASC),
               INDEX i1 (z ASC) PARTITION BY LIST (z) (
                 PARTITION p1 VALUES IN ((1), (2)),
                 PARTITION p2 VALUES IN ((3), (4))
               ),
               INDEX i2 (w ASC) PARTITION BY LIST (w) (
                 PARTITION p3 VALUES IN ((5), (6)),
                 PARTITION p4 VALUES IN ((7), (8))
               ),
               FAMILY fam_0_x_y_z_w (x, y, z, w)
            );
            ALTER PARTITION p3 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15411;
            ALTER PARTITION p4 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15418;
            ALTER PARTITION p1 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15210;
            ALTER PARTITION p2 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15213
--- done: testdata/logic_test/zone with config 5node: 144 tests, 1 failures
            --- FAIL: TestCCLLogic/5node/zone/authorization (1.34s)
=== RUN   TestCCLLogic/5node/zone
=== PAUSE TestCCLLogic/5node/zone
=== CONT  TestCCLLogic/5node/zone
        --- FAIL: TestCCLLogic/5node/zone (4.21s)
=== RUN   TestCCLLogic/5node
    --- FAIL: TestCCLLogic/5node (0.00s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestCCLLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/logictestccl).TestCCLLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2628995&tab=buildLog) on [master@70e72e9b833c763fd933cd752283d556d1faa785](https://github.com/cockroachdb/cockroach/commits/70e72e9b833c763fd933cd752283d556d1faa785):

```
               ),
               INDEX i2 (w ASC) PARTITION BY LIST (w) (
                 PARTITION p3 VALUES IN ((5), (6)),
                 PARTITION p4 VALUES IN ((7), (8))
               ),
               FAMILY fam_0_x_y_z_w (x, y, z, w)
            );
            ALTER PARTITION p1 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15210;
            ALTER PARTITION p2 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15213;
            ALTER PARTITION p3 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15411;
            ALTER PARTITION p4 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15418
        but found (query options: "") :
            t  CREATE TABLE public.t (
               x INT8 NOT NULL,
               y INT8 NOT NULL,
               z INT8 NULL,
               w INT8 NULL,
               CONSTRAINT "primary" PRIMARY KEY (y ASC),
               UNIQUE INDEX t_x_key (x ASC),
               INDEX i1 (z ASC) PARTITION BY LIST (z) (
                 PARTITION p1 VALUES IN ((1), (2)),
                 PARTITION p2 VALUES IN ((3), (4))
               ),
               INDEX i2 (w ASC) PARTITION BY LIST (w) (
                 PARTITION p3 VALUES IN ((5), (6)),
                 PARTITION p4 VALUES IN ((7), (8))
               ),
               FAMILY fam_0_x_y_z_w (x, y, z, w)
            );
            ALTER PARTITION p3 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15411;
            ALTER PARTITION p4 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15418;
            ALTER PARTITION p1 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15210;
            ALTER PARTITION p2 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15213
--- done: testdata/logic_test/zone with config 5node: 144 tests, 1 failures
--- total progress: 579 statements/queries
            --- FAIL: TestCCLLogic/5node/zone/authorization (2.18s)
=== RUN   TestCCLLogic/5node/zone
=== PAUSE TestCCLLogic/5node/zone
=== CONT  TestCCLLogic/5node/zone
        --- FAIL: TestCCLLogic/5node/zone (5.34s)
=== RUN   TestCCLLogic/5node
    --- FAIL: TestCCLLogic/5node (0.00s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestCCLLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/logictestccl).TestCCLLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2629172&tab=buildLog) on [master@70e72e9b833c763fd933cd752283d556d1faa785](https://github.com/cockroachdb/cockroach/commits/70e72e9b833c763fd933cd752283d556d1faa785):

```
               ),
               INDEX i2 (w ASC) PARTITION BY LIST (w) (
                 PARTITION p3 VALUES IN ((5), (6)),
                 PARTITION p4 VALUES IN ((7), (8))
               ),
               FAMILY fam_0_x_y_z_w (x, y, z, w)
            );
            ALTER PARTITION p1 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15210;
            ALTER PARTITION p2 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15213;
            ALTER PARTITION p3 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15411;
            ALTER PARTITION p4 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15418
        but found (query options: "") :
            t  CREATE TABLE public.t (
               x INT8 NOT NULL,
               y INT8 NOT NULL,
               z INT8 NULL,
               w INT8 NULL,
               CONSTRAINT "primary" PRIMARY KEY (y ASC),
               UNIQUE INDEX t_x_key (x ASC),
               INDEX i1 (z ASC) PARTITION BY LIST (z) (
                 PARTITION p1 VALUES IN ((1), (2)),
                 PARTITION p2 VALUES IN ((3), (4))
               ),
               INDEX i2 (w ASC) PARTITION BY LIST (w) (
                 PARTITION p3 VALUES IN ((5), (6)),
                 PARTITION p4 VALUES IN ((7), (8))
               ),
               FAMILY fam_0_x_y_z_w (x, y, z, w)
            );
            ALTER PARTITION p3 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15411;
            ALTER PARTITION p4 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15418;
            ALTER PARTITION p1 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15210;
            ALTER PARTITION p2 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15213
--- done: testdata/logic_test/zone with config 5node: 144 tests, 1 failures
--- total progress: 579 statements/queries
            --- FAIL: TestCCLLogic/5node/zone/authorization (2.17s)
=== RUN   TestCCLLogic/5node/zone
=== PAUSE TestCCLLogic/5node/zone
=== CONT  TestCCLLogic/5node/zone
        --- FAIL: TestCCLLogic/5node/zone (5.68s)
=== RUN   TestCCLLogic/5node
    --- FAIL: TestCCLLogic/5node (0.00s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestCCLLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/logictestccl).TestCCLLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2629217&tab=buildLog) on [master@ac799bd8f330f47b08b8e0958e3ede0e542b2acc](https://github.com/cockroachdb/cockroach/commits/ac799bd8f330f47b08b8e0958e3ede0e542b2acc):

```
               ),
               INDEX i2 (w ASC) PARTITION BY LIST (w) (
                 PARTITION p3 VALUES IN ((5), (6)),
                 PARTITION p4 VALUES IN ((7), (8))
               ),
               FAMILY fam_0_x_y_z_w (x, y, z, w)
            );
            ALTER PARTITION p1 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15210;
            ALTER PARTITION p2 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15213;
            ALTER PARTITION p3 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15411;
            ALTER PARTITION p4 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15418
        but found (query options: "") :
            t  CREATE TABLE public.t (
               x INT8 NOT NULL,
               y INT8 NOT NULL,
               z INT8 NULL,
               w INT8 NULL,
               CONSTRAINT "primary" PRIMARY KEY (y ASC),
               UNIQUE INDEX t_x_key (x ASC),
               INDEX i1 (z ASC) PARTITION BY LIST (z) (
                 PARTITION p1 VALUES IN ((1), (2)),
                 PARTITION p2 VALUES IN ((3), (4))
               ),
               INDEX i2 (w ASC) PARTITION BY LIST (w) (
                 PARTITION p3 VALUES IN ((5), (6)),
                 PARTITION p4 VALUES IN ((7), (8))
               ),
               FAMILY fam_0_x_y_z_w (x, y, z, w)
            );
            ALTER PARTITION p3 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15411;
            ALTER PARTITION p4 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15418;
            ALTER PARTITION p1 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15210;
            ALTER PARTITION p2 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15213
--- done: testdata/logic_test/zone with config 5node: 144 tests, 1 failures
--- total progress: 579 statements/queries
            --- FAIL: TestCCLLogic/5node/zone/authorization (2.93s)
=== RUN   TestCCLLogic/5node/zone
=== PAUSE TestCCLLogic/5node/zone
=== CONT  TestCCLLogic/5node/zone
        --- FAIL: TestCCLLogic/5node/zone (6.49s)
=== RUN   TestCCLLogic/5node
    --- FAIL: TestCCLLogic/5node (0.00s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestCCLLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/logictestccl).TestCCLLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2629283&tab=buildLog) on [master@171cc86d6d84c110197485b5d886d77cddc12399](https://github.com/cockroachdb/cockroach/commits/171cc86d6d84c110197485b5d886d77cddc12399):

```
               INDEX i2 (w ASC) PARTITION BY LIST (w) (
                 PARTITION p3 VALUES IN ((5), (6)),
                 PARTITION p4 VALUES IN ((7), (8))
               ),
               FAMILY fam_0_x_y_z_w (x, y, z, w)
            );
            ALTER PARTITION p1 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15210;
            ALTER PARTITION p2 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15213;
            ALTER PARTITION p3 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15411;
            ALTER PARTITION p4 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15418
        but found (query options: "") :
            t  CREATE TABLE public.t (
               x INT8 NOT NULL,
               y INT8 NOT NULL,
               z INT8 NULL,
               w INT8 NULL,
               CONSTRAINT "primary" PRIMARY KEY (y ASC),
               UNIQUE INDEX t_x_key (x ASC),
               INDEX i1 (z ASC) PARTITION BY LIST (z) (
                 PARTITION p1 VALUES IN ((1), (2)),
                 PARTITION p2 VALUES IN ((3), (4))
               ),
               INDEX i2 (w ASC) PARTITION BY LIST (w) (
                 PARTITION p3 VALUES IN ((5), (6)),
                 PARTITION p4 VALUES IN ((7), (8))
               ),
               FAMILY fam_0_x_y_z_w (x, y, z, w)
            );
            ALTER PARTITION p3 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15411;
            ALTER PARTITION p4 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15418;
            ALTER PARTITION p1 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15210;
            ALTER PARTITION p2 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15213
--- done: testdata/logic_test/zone with config 5node: 144 tests, 1 failures
E210130 00:32:15.134316 66425 sql/stats/automatic_stats.go:302  [-] 4  failed to refresh stats: node unavailable; try another peer
--- total progress: 579 statements/queries
            --- FAIL: TestCCLLogic/5node/zone/authorization (2.13s)
=== RUN   TestCCLLogic/5node/zone
=== PAUSE TestCCLLogic/5node/zone
=== CONT  TestCCLLogic/5node/zone
        --- FAIL: TestCCLLogic/5node/zone (5.57s)
=== RUN   TestCCLLogic/5node
    --- FAIL: TestCCLLogic/5node (0.00s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestCCLLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/logictestccl).TestCCLLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2629500&tab=buildLog) on [master@171cc86d6d84c110197485b5d886d77cddc12399](https://github.com/cockroachdb/cockroach/commits/171cc86d6d84c110197485b5d886d77cddc12399):

```
               w INT8 NULL,
               CONSTRAINT "primary" PRIMARY KEY (y ASC),
               UNIQUE INDEX t_x_key (x ASC),
               INDEX i1 (z ASC) PARTITION BY LIST (z) (
                 PARTITION p1 VALUES IN ((1), (2)),
                 PARTITION p2 VALUES IN ((3), (4))
               ),
               INDEX i2 (w ASC) PARTITION BY LIST (w) (
                 PARTITION p3 VALUES IN ((5), (6)),
                 PARTITION p4 VALUES IN ((7), (8))
               ),
               FAMILY fam_0_x_y_z_w (x, y, z, w)
            );
            ALTER PARTITION p1 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15210;
            ALTER PARTITION p2 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15213;
            ALTER PARTITION p3 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15411;
            ALTER PARTITION p4 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15418
        but found (query options: "") :
            t  CREATE TABLE public.t (
               x INT8 NOT NULL,
               y INT8 NOT NULL,
               z INT8 NULL,
               w INT8 NULL,
               CONSTRAINT "primary" PRIMARY KEY (y ASC),
               UNIQUE INDEX t_x_key (x ASC),
               INDEX i1 (z ASC) PARTITION BY LIST (z) (
                 PARTITION p1 VALUES IN ((1), (2)),
                 PARTITION p2 VALUES IN ((3), (4))
               ),
               INDEX i2 (w ASC) PARTITION BY LIST (w) (
                 PARTITION p3 VALUES IN ((5), (6)),
                 PARTITION p4 VALUES IN ((7), (8))
               ),
               FAMILY fam_0_x_y_z_w (x, y, z, w)
            );
            ALTER PARTITION p3 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15411;
            ALTER PARTITION p4 OF INDEX test.public.t@i2 CONFIGURE ZONE USING
              gc.ttlseconds = 15418;
            ALTER PARTITION p1 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15210;
            ALTER PARTITION p2 OF INDEX test.public.t@i1 CONFIGURE ZONE USING
              gc.ttlseconds = 15213
--- done: testdata/logic_test/zone with config 5node: 144 tests, 1 failures
--- total progress: 579 statements/queries
            --- FAIL: TestCCLLogic/5node/zone/authorization (1.16s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestCCLLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row
+++ b/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row
@@ -300,6 +300,10 @@ CREATE TABLE public.regional_by_row_table (
   UNIQUE INDEX unique_b_a (b ASC, a ASC),
   FAMILY fam_0_pk_pk2_a_b_crdb_region (pk, pk2, a, b, crdb_region)
 ) LOCALITY REGIONAL BY ROW;
+ALTER PARTITION "ap-southeast-2" OF INDEX multi_region_test_db.public.regional_by_row_table@new_idx CONFIGURE ZONE USING
+  num_replicas = 3,
+  constraints = '{+region=ap-southeast-2: 1}',
+  lease_preferences = '[[+region=ap-southeast-2]]';
 ALTER PARTITION "ap-southeast-2" OF INDEX multi_region_test_db.public.regional_by_row_table@primary CONFIGURE ZONE USING
   num_replicas = 3,
   constraints = '{+region=ap-southeast-2: 1}',
@@ -312,6 +316,14 @@ ALTER PARTITION "ap-southeast-2" OF INDEX multi_region_test_db.public.regional_b
   num_replicas = 3,
   constraints = '{+region=ap-southeast-2: 1}',
   lease_preferences = '[[+region=ap-southeast-2]]';
+ALTER PARTITION "ap-southeast-2" OF INDEX multi_region_test_db.public.regional_by_row_table@unique_b_a CONFIGURE ZONE USING
+  num_replicas = 3,
+  constraints = '{+region=ap-southeast-2: 1}',
+  lease_preferences = '[[+region=ap-southeast-2]]';
+ALTER PARTITION "ca-central-1" OF INDEX multi_region_test_db.public.regional_by_row_table@new_idx CONFIGURE ZONE USING
+  num_replicas = 3,
+  constraints = '{+region=ca-central-1: 1}',
+  lease_preferences = '[[+region=ca-central-1]]';
 ALTER PARTITION "ca-central-1" OF INDEX multi_region_test_db.public.regional_by_row_table@primary CONFIGURE ZONE USING
   num_replicas = 3,
   constraints = '{+region=ca-central-1: 1}',
@@ -324,6 +336,14 @@ ALTER PARTITION "ca-central-1" OF INDEX multi_region_test_db.public.regional_by_
   num_replicas = 3,
   constraints = '{+region=ca-central-1: 1}',
   lease_preferences = '[[+region=ca-central-1]]';
+ALTER PARTITION "ca-central-1" OF INDEX multi_region_test_db.public.regional_by_row_table@unique_b_a CONFIGURE ZONE USING
+  num_replicas = 3,
+  constraints = '{+region=ca-central-1: 1}',
+  lease_preferences = '[[+region=ca-central-1]]';
+ALTER PARTITION "us-east-1" OF INDEX multi_region_test_db.public.regional_by_row_table@new_idx CONFIGURE ZONE USING
+  num_replicas = 3,
+  constraints = '{+region=us-east-1: 1}',
+  lease_preferences = '[[+region=us-east-1]]';
 ALTER PARTITION "us-east-1" OF INDEX multi_region_test_db.public.regional_by_row_table@primary CONFIGURE ZONE USING
   num_replicas = 3,
   constraints = '{+region=us-east-1: 1}',
@@ -336,26 +356,6 @@ ALTER PARTITION "us-east-1" OF INDEX multi_region_test_db.public.regional_by_row
   num_replicas = 3,
   constraints = '{+region=us-east-1: 1}',
   lease_preferences = '[[+region=us-east-1]]';
-ALTER PARTITION "ap-southeast-2" OF INDEX multi_region_test_db.public.regional_by_row_table@new_idx CONFIGURE ZONE USING
-  num_replicas = 3,
-  constraints = '{+region=ap-southeast-2: 1}',
-  lease_preferences = '[[+region=ap-southeast-2]]';
-ALTER PARTITION "ca-central-1" OF INDEX multi_region_test_db.public.regional_by_row_table@new_idx CONFIGURE ZONE USING
-  num_replicas = 3,
-  constraints = '{+region=ca-central-1: 1}',
-  lease_preferences = '[[+region=ca-central-1]]';
-ALTER PARTITION "us-east-1" OF INDEX multi_region_test_db.public.regional_by_row_table@new_idx CONFIGURE ZONE USING
-  num_replicas = 3,
-  constraints = '{+region=us-east-1: 1}',
-  lease_preferences = '[[+region=us-east-1]]';
-ALTER PARTITION "ap-southeast-2" OF INDEX multi_region_test_db.public.regional_by_row_table@unique_b_a CONFIGURE ZONE USING
-  num_replicas = 3,
-  constraints = '{+region=ap-southeast-2: 1}',
-  lease_preferences = '[[+region=ap-southeast-2]]';
-ALTER PARTITION "ca-central-1" OF INDEX multi_region_test_db.public.regional_by_row_table@unique_b_a CONFIGURE ZONE USING
-  num_replicas = 3,
-  constraints = '{+region=ca-central-1: 1}',
-  lease_preferences = '[[+region=ca-central-1]]';
 ALTER PARTITION "us-east-1" OF INDEX multi_region_test_db.public.regional_by_row_table@unique_b_a CONFIGURE ZONE USING
   num_replicas = 3,
   constraints = '{+region=us-east-1: 1}',

--- a/pkg/sql/delegate/show_table.go
+++ b/pkg/sql/delegate/show_table.go
@@ -24,7 +24,7 @@ func (d *delegator) delegateShowCreate(n *tree.ShowCreate) (tree.Statement, erro
 
 	const showCreateQuery = `
 WITH zone_configs AS (
-    SELECT string_agg(raw_config_sql, e';\n') FROM crdb_internal.zones
+    SELECT string_agg(raw_config_sql, e';\n' ORDER BY partition_name, index_name) FROM crdb_internal.zones
     WHERE database_name = %[1]s
     AND table_name = %[2]s
     AND raw_config_yaml IS NOT NULL

